streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -606,6 +606,25 @@ public void onAssignment(Assignment assignment) {
             }
         }
         metadataWithInternalTopics = Cluster.empty().withPartitions(topicToPartitionInfo);
+        updateSubscribedTopics(assignment);
+    }
+
+
+    private void updateSubscribedTopics(Assignment assignment) {
+        if (streamThread != null && streamThread.builder.sourceTopicPattern() != null) {
```

Comment:
you're right it can't, I put that in for unit tests, I'll remove and update the tests